### ISSUE TITLE
Support v0.8.4 custom errors

### ISSUE THREAD

**Issue by paulrberg** [sc-forks/solidity-coverage#639]
See the announcement [article](https://blog.soliditylang.org/2021/04/21/custom-errors/) for the new custom error syntax.

For the following code:

```solidity
error GreeterError();

contract Greeter {
    function throwError() external pure {
        revert GreeterError();
    }
}
```

I get this error when running `yarn coverage`:

> Error in plugin solidity-coverage: Error: Could not instrument: Greeter.sol. (Please verify solc can compile this file without errors.) missing 'constant' at 'GreeterError' (6:6)

Perhaps the fix for this is as easy as bumping `@solidity-parser/parser` to the latest version (which is v0.13.2, at the time of opening this issue)?

**Comment by paulrberg** [comment]
Yup, the fix is easy as bumping that dependency version. Before @cgewecke manages to find the time to release a new version of `solidity-coverage`, we can add this to our `package.json`:

```json
...
"resolutions": {
  "@solidity-parser/parser": "^0.13.2"
},
...
```

The [resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) feature of yarn force-installs a specific dependency (B) version even if another dependency (A) depends on another version of that dependency (B).

**Comment by lepidotteri** [comment]
@cgewecke can you update the solc parser to support v0.8.x too?

**Comment by paulrberg** [comment]
> @cgewecke can you update the solc parser to support v0.8.x too?

Related: #626.

**Comment by julianmrodri** [comment]
Hi! any news about this? Will this be supported shortly?

**Comment by lepidotteri** [comment]
pussies

**Comment by paulrberg** [comment]
> pussies

@cgewecke is not compensated for maintaining solidity-coverage. If you need urgent support of v0.8.4 custom errors in your coverage report, you might want to implement them yourself and make a PR.

**Comment by lepidotteri** [comment]
> > pussies
> 
> @cgewecke is not compensated for maintaining solidity-coverage. If you need urgent support of v0.8.4 custom errors in your coverage report, you might want to implement them yourself and make a PR.

It doesn't even work with Hardhat 2.5 if the hardfork is London, it works if selecting a prior hardfork; 

I get he's not paid, but he should fucking be; this is the only/best coverage tool for the whole of Solidity and you just cant afford to let this kind of tools get behind; same thing with Waffle and likely others, its a disgrace how disorganized things are still run with regards to developer ecosystem readiness.
